aws-android-sdk-s3/src/test/java/com/amazonaws/mobileconnectors/s3/transferutility/UploadInputStreamTest.java

```diff
@@ -0,0 +1,221 @@
+        verify(transferUtility, times(1)).upload(eq(DEFAULT_BUCKET), eq(key),
+                any(File.class), eq(metadata), isNull(CannedAccessControlList.class), isNull(TransferListener.class));
+    }
+
+    @Test
+    public void testUploadWithCannedAcl() throws IOException {
+        String key = getRandomString();
+        transferUtility.upload(key, inputStream, UploadOptions.builder().cannedAcl(cannedAcl).build());
+        verify(transferUtility, times(1)).upload(eq(DEFAULT_BUCKET), eq(key),
+                any(File.class), any(ObjectMetadata.class), eq(cannedAcl), isNull(TransferListener.class));
+    }
+
+    @Test
+    public void testUploadWithMetadataAndCannedAcl() throws IOException {
+        String key = getRandomString();
+        transferUtility.upload(key, inputStream,
+                UploadOptions.builder().objectMetadata(metadata).cannedAcl(cannedAcl).build());
+        verify(transferUtility, times(1)).upload(eq(DEFAULT_BUCKET), eq(key),
+                any(File.class), eq(metadata), eq(cannedAcl), isNull(TransferListener.class));
+    }
+
+    @Test
+    public void testUploadWithBucketAndMetadata() throws IOException {
+        String key = getRandomString();
+        String bucket = getRandomString();
+        transferUtility.upload(key, inputStream,
+                UploadOptions.builder().bucket(bucket).objectMetadata(metadata).build());
+        verify(transferUtility, times(1)).upload(eq(bucket), eq(key), any(File.class),
+                eq(metadata), isNull(CannedAccessControlList.class), isNull(TransferListener.class));
+    }
+
+    @Test
+    public void testUploadWithBucketAndCannedAcl() throws IOException {
+        String key = getRandomString();
+        String bucket = getRandomString();
+        transferUtility.upload(key, inputStream,
+                UploadOptions.builder().bucket(bucket).cannedAcl(cannedAcl).build());
+        verify(transferUtility, times(1)).upload(eq(bucket), eq(key), any(File.class),
+                any(ObjectMetadata.class), eq(cannedAcl), isNull(TransferListener.class));
+    }
+
+    @Test
+    public void testUploadWithBucketAndMetadataAndCannedAcl() throws IOException {
+        String key = getRandomString();
+        String bucket = getRandomString();
+        transferUtility.upload(key, inputStream,
+                UploadOptions.builder().bucket(bucket).objectMetadata(metadata).cannedAcl(cannedAcl).build());
+        verify(transferUtility, times(1)).upload(eq(bucket), eq(key), any(File.class),
+                eq(metadata), eq(cannedAcl), isNull(TransferListener.class));
+    }
+
+    @Test
+    public void testUploadWithMetadataAndCannedAclAndListener() throws IOException {
+        String key = getRandomString();
+        transferUtility.upload(key, inputStream,
+                UploadOptions.builder().objectMetadata(metadata).cannedAcl(cannedAcl).transferListener(listener).build());
```

Comment:
There's not really an issue with the way you have it, but I'd suggest putting each step of the Builder on a new line, as it's just a little bit easier to read that way, and sort of the standard way of using a Builder, like this:

```suggestion
        transferUtility.upload(key, inputStream, UploadOptions.builder()
                .objectMetadata(metadata)
                .cannedAcl(cannedAcl)
                .transferListener(listener)
                .build());
```

Same suggestion for the other places you are using the Builder, in the other tests.